### DISCUSSION TITLE
Flask 2.0 Async Not working

### DISCUSSION THREAD

**Discussion by csharmatech** [pallets/flask#4896] [Q&A] [answered]
Hi,
I wanted to explore the async feature and hence installed async on Flask 2.0 and defined a simple async route as below, however when accessing the URL, it throws errors as mentioned below.

This is my async route:

```
@app.route('/home')
async def myindex():
    await asyncio.sleep(1)
    return "Hello"

```

I am using gunicorn with websocket support(geventwebsocket) and below are the installed packages:

asyncio==3.4.3
Flask==2.0.3
Flask-SocketIO==5.3.1
greenlet==1.1.3.post0

Errors while accessing "/home" as below:

```
   CGroup: /system.slice/myapp.service
           ├─ 976 /usr/bin/python3.6 /usr/local/bin/gunicorn -k geventwebsocket.gunicorn.workers.GeventWebSocketWorker -w 1 myapp:app --bind 0.0.0.0:5000
           └─1428 /usr/bin/python3.6 /usr/local/bin/gunicorn -k geventwebsocket.gunicorn.workers.GeventWebSocketWorker -w 1 myapp:app --bind 0.0.0.0:5000

Dec 04 23:05:30 Flask gunicorn[1428]:     rv = self.handle_user_exception(e)
Dec 04 23:05:30 Flask gunicorn[1428]:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1516, in full_dispatch_request
Dec 04 23:05:30 Flask gunicorn[1428]:     rv = self.dispatch_request()
Dec 04 23:05:30 Flask gunicorn[1428]:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1502, in dispatch_request
Dec 04 23:05:30 Flask gunicorn[1428]:     return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
Dec 04 23:05:30 Flask gunicorn[1428]:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1599, in ensure_sync
Dec 04 23:05:30 Flask gunicorn[1428]:     return self.async_to_sync(func)
Dec 04 23:05:30 Flask gunicorn[1428]:   File "/usr/local/lib/python3.6/site-packages/flask/app.py", line 1627, in async_to_sync
Dec 04 23:05:30 Flask gunicorn[1428]:     "Async cannot be used with this combination of Python "
Dec 04 23:05:30 Flask gunicorn[1428]: RuntimeError: Async cannot be used with this combination of Python and Greenlet versions.

```

**Comment by csharmatech** [comment]
JFYI, my python version is 3.7.1

**Reply by ThiefMaster** [reply]
You are using 3.6 according to your error. But even 3.7 is ancient and not a version you should be using.

**Reply by ThiefMaster** [reply]
Make sure you use a recent Python version and also recent versions of Flask etc. Error reports using old versions are not particularly helpful.

**Reply by csharmatech** [reply]
How do I install Flask such that it use the latest Python?

**Reply by csharmatech** [reply]
If I uninstall and re-install Flask, it again defaults to Python 3.6

**Reply by ThiefMaster** [reply]
Make sure your correct Python version is active (e.g. by using a virtualenv that has the right version - I have no clue on what system you are and how you isntalled Python), then you can simply do `pip install flask` to get the latest Flask.

**Reply by csharmatech** [reply]
Ok, I installed the latest Python 3.9 and also latest Flask and now its working.

**Comment by ThiefMaster** [comment, marked as answer]
Use a Python version that's not ancient (3.6 is even EOL).
